### PR TITLE
Test for changing BhpLimit in history mode using weltarg

### PR DESCRIPTION
**Pull request by totto82**
This test does not currently work but points to an issue with the current handling of weltarg in combination with historical mode. The test is based on my interpretation of the manual and tests in Eclipse.

### CODE REVIEW

**Comment by joakim-hove** [comment]
First of all - let me congratulate you with your first(?) *test first experience*; now you have a failing test - just hack away until the test passes - and wham bang you are done - never to see the bug again!

I would assume that the first `BOOST_CHECK()` statements are ok? But I agree that resetting of default BHP limit when changing between injector and producer is not implemented - at all. Is the behavior you ask for in the test - what you consider the Right&trade; way to do it?

File: opm/parser/eclipse/EclipseState/Schedule/tests/ScheduleTests.cpp [hunk 1]
```diff
@@ -1347,3 +1347,80 @@ BOOST_AUTO_TEST_CASE(createDeckWithOutOilVaporizationProperties) {
...
+    DeckPtr deck = parser.parseString(input, parseMode);
+    std::shared_ptr<const EclipseGrid> grid = std::make_shared<const EclipseGrid>(10, 10, 10);
+    IOConfigPtr ioConfig;
+    Schedule schedule(parseMode , grid, deck, ioConfig);
+    WellPtr well_p = schedule.getWell("P");
+
+    BOOST_CHECK_EQUAL(well_p->getProductionProperties(0).BHPLimit, 0); //start
+    BOOST_CHECK_EQUAL(well_p->getProductionProperties(1).BHPLimit, 50 * 1e5); // 1
+    // The BHP limit should not be effected by WCONHIST
+    BOOST_CHECK_EQUAL(well_p->getProductionProperties(2).BHPLimit, 50 * 1e5); // 2
```
Comment (totto82) [inline opm/parser/eclipse/EclipseState/Schedule/tests/ScheduleTests.cpp]: This fails

**Comment by totto82** [comment]
> Is the behavior you ask for in the test - what you consider the Right™ way to do it?
Yes. 

I believe that the flowing changes will fix the two failures
1) Don't reset the bhp limit every WCONHIST
2) Reset the bhp limit when moving between producer and injector.

**Comment by joakim-hove** [comment]
See: https://github.com/OPM/opm-parser/pull/621

**Pull request closed without merging**